### PR TITLE
add read-only website api for collections

### PR DESCRIPTION
**Pull request by gmethvin**

### CODE REVIEW

File: shoebox/app/com/keepit/controllers/website/BookmarksController.scala [hunk 1]
```diff
@@ -41,13 +50,22 @@ class BookmarksController @Inject() (
     db.readOnly { implicit s => ExternalId.asOpt[Bookmark](id).flatMap(bookmarkRepo.getOpt) } map (_.externalId)
   }
 
-  def allKeeps(before: Option[String], after: Option[String], count: Int) = AuthenticatedJsonAction { request =>
-    (before map getBookmarkExternalId, after map getBookmarkExternalId) match {
-      case (Some(None), _) => BadRequest(s"Invalid id for before: ${before.get}")
-      case (_, Some(None)) => BadRequest(s"Invalid id for after: ${after.get}")
-      case (beforeId, afterId) => Async {
+  private def getCollectionByExternalId(userId: Id[User], id: String): Option[Collection] = {
```
Comment (2is10) [inline shoebox/app/com/keepit/controllers/website/BookmarksController.scala:53]: how about putting this in `CollectionRepo`?

File: shoebox/app/com/keepit/model/Bookmark.scala [hunk 2]
```diff
@@ -193,7 +194,13 @@ class BookmarkRepoImpl @Inject() (
...
           b.createdAt > after.createdAt || b.id > after.id.get && b.createdAt === after.createdAt
         } getOrElse (b.id === b.id))
-    } yield b).sortBy(_.id desc).sortBy(_.createdAt desc).take(count).list
+    } yield b)
+    (collectionId.map { cid =>
+      for {
+        (b, ktc) <- q join keepToCollectionRepo.table on (_.id === _.bookmarkId)
+          if ktc.collectionId === cid
+      } yield b
+    } getOrElse q).sortBy(_.id desc).sortBy(_.createdAt desc).take(count).list
```
Comment (2is10) [inline shoebox/app/com/keepit/model/Bookmark.scala:203]: Nice!

**Comment by 2is10** [comment]
lgtm

File: shoebox/app/com/keepit/controllers/website/BookmarksController.scala (hunk 1 above)
Comment (gmethvin) [inline shoebox/app/com/keepit/controllers/website/BookmarksController.scala:53]: `CollectionRepo` should probably have a `getByUserAndExternalId`.

**Pull request merged**